notification/src/config/init/ensure-adyen-webhook.js

```diff
@@ -59,41 +60,57 @@ async function ensureAdyenWebhook(adyenApiKey, webhookUrl, merchantId) {
     const webhookId = createWebhookResponseJson.id
 
     logger.info(`New webhook was created with ID ${webhookId}`)
+    return webhookId
   } catch (err) {
     throw Error(
       `Failed to ensure adyen webhook for project ${merchantId}.` +
         `Error: ${JSON.stringify(serializeError(err))}`
     )
   }
+}
+
+async function ensureAdyenHmac(adyenApiKey, merchantId, webhookId) {
+  const generateHmacResponse = await fetch(
+    `https://management-test.adyen.com/v1/merchants/${merchantId}/webhooks/${webhookId}/generateHmac`,
+    {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Api-Key': adyenApiKey,
+      },
+    }
+  )
 
-  // todo: will be done later
-  // const generateHmacResponse = await fetch(
-  //   `https://management-test.adyen.com/v1/merchants/${merchantId}/webhooks/${webhookId}/generateHmac`, {
-  //   method: 'POST',
-  //   headers: {
-  //     'Content-Type': 'application/json',
-  //     'X-Api-Key': adyenApiKey
-  //   },
-  // })
-  //
-  // const generateHmacResponseJson = await generateHmacResponse.json()
-  // const { hmacKey } = generateHmacResponseJson
-  //
-  // logger.info(`HMAC key generated ${hmacKey}`)
+  const generateHmacResponseJson = await generateHmacResponse.json()
+  const { hmacKey } = generateHmacResponseJson
+
+  return hmacKey
 }
 
 async function ensureAdyenWebhooksForAllProjects() {
   const adyenMerchantAccounts = config.getAllAdyenMerchantAccounts()
+  const jsonConfig = loadConfig()
   for (const adyenMerchantId of adyenMerchantAccounts) {
     const adyenConfig = config.getAdyenConfig(adyenMerchantId)
     if (adyenConfig.notificationBaseUrl) {
-      await ensureAdyenWebhook(
+      const webhookId = await ensureAdyenWebhook(
         adyenConfig.apiKey,
         adyenConfig.notificationBaseUrl,
         adyenMerchantId
       )
+      if (adyenConfig.enableHmacSignature && !adyenConfig.secretHmacKey) {
+        const hmacKey = await ensureAdyenHmac(
+          adyenConfig.apiKey,
+          adyenMerchantId,
+          webhookId
+        )
+        jsonConfig.adyen[adyenMerchantId].secretHmacKey = hmacKey
+      }
     }
   }
+
+  mainLogger.info('Set the following environmental variable')
+  console.log(`ADYEN_INTEGRATION_CONFIG='${JSON.stringify(jsonConfig)}'`)
```

Comment:
is it possible to update the environment variable for user here instead of printing it out?

Reply:
Not sure. Can you make a suggestion how to do that?

Reply:
> Not sure. Can you make a suggestion how to do that?

e.g. `process.env.ADYEN_INTEGRATION_CONFIG = jsonConfig`

Reply:
> 

In case we need to let user know the generated HMAC key, we can update ADYEN_INTEGRATION_CONFIG in environment variable first and then print it out. wdyt?

Reply:
> One thing to be aware of: HMAC key is kind of CTP client secret - once it is generated, you cannot get the same key anymore if you refresh the page. Same idea with HMAC key - you can only generate new HMAC keys, not get the existing one from Adyen.

If we need to let user know the generated HMAC key since it cannot be retrieved later, we can update `ADYEN_INTEGRATION_CONFIG` in environment variable first and then print it out. wdyt?

Reply:
This means I have to run this HMAC generation before the start the notification module, right? Because right now this update runs separately and manually by the user. And how do you ensure that you always have the same HMAC key even after restart?